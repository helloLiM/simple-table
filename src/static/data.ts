export const COLUMN = [{
    key: 'id',
    label: 'ID',
    hidden: 'false',
    sortable: true,
  },{
    key: 'name',
    label: '名称',
    hidden: 'false',
    sortable: true,
  },{
    key: 'sex',
    label: '性别',
    hidden: 'false',
  },{
    key: 'age',
    label: '年龄',
    hidden: 'true',
    sortable: true,
  }]
export const DATA = [
    {
        id: '0',
        name: 'a',
        sex: 'male',
        age: 12,
    }, {
        id: '1',
        name: 'b',
        sex: 'female',
        age: 64,
    }, {
        id: '2',
        name: 'c',
        sex: 'male',
        age: 5,
    }, {
        id: '3',
        name: 'd',
        sex: 'female',
        age: 21,
    }, {
        id: '4',
        name: 'e',
        sex: 'female',
        age: 75,
    }, {
        id: '5',
        name: 'f',
        sex: 'male',
        age: 25,
    }, {
        id: '6',
        name: 'g',
        sex: 'male',
        age: 42,
    }, {
        id: '8',
        name: 'h',
        sex: 'female',
        age: 13,
    },{
        id: '9',
        name: 'wv',
        sex: 'female',
        age: 56,
    },{
        id: '10',
        name: 'ki',
        sex: 'male',
        age: 76,
    },{
        id: '11',
        name: 'xe',
        sex: 'female',
        age: 24,
    },{
        id: '12',
        name: 'yu',
        sex: 'female',
        age: 18,
    },{
        id: '13',
        name: 'sd',
        sex: 'male',
        age: 42,
    },{
        id: '14',
        name: 'vb',
        sex: 'female',
        age: 23,
    },{
        id: '15',
        name: 'dg',
        sex: 'male',
        age: 89,
    },{
        id: '16',
        name: 'nb',
        sex: 'female',
        age: 67,
    },{
        id: '17',
        name: 'fh',
        sex: 'female',
        age: 4,
    },{
        id: '18',
        name: 'ty',
        sex: 'female',
        age: 96,
    },{
        id: '19',
        name: 'ui',
        sex: 'male',
        age: 9,
    },{
        id: '20',
        name: 'mj',
        sex: 'male',
        age: 74,
    },{
        id: '21',
        name: 'pi',
        sex: 'female',
        age: 34,
    },{
        id: '22',
        name: 'ol',
        sex: 'female',
        age: 34,
    },{
        id: '23',
        name: 'asd',
        sex: 'male',
        age: 84,
    },{
        id: '24',
        name: 'az',
        sex: 'male',
        age: 46,
    },{
        id: '25',
        name: 'mu',
        sex: 'female',
        age: 57,
    },{
        id: '26',
        name: 'mn',
        sex: 'male',
        age: 73,
    },{
        id: '27',
        name: 'hg',
        sex: 'female',
        age: 98,
    },{
        id: '28',
        name: 'bg',
        sex: 'female',
        age: 42,
    },{
        id: '29',
        name: 'vcd',
        sex: 'male',
        age: 3,
    },{
        id: '30',
        name: 'qw',
        sex: 'female',
        age: 33,
    },{
        id: '31',
        name: 'y6',
        sex: 'female',
        age: 34,
    },{
        id: '32',
        name: 'wer8',
        sex: 'male',
        age: 42,
    },{
        id: '33',
        name: 'fyj',
        sex: 'female',
        age: 14,
    }
];