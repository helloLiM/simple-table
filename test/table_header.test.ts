import { shallowMount,mount } from "@vue/test-utils";
import { expect, test } from "vitest";
import tableHeader from "../src/components/table_header";
import { nextTick } from 'vue';
import * as _ from 'lodash';
import { DATA, COLUMN } from "../src/static/data"


test("mount component", () => {
  const wrapper:any = shallowMount(tableHeader, {
    props: {
      columns : COLUMN,
      sortParams :  {
        field: '',
        direction: '',
      },
    },
  });
  expect(wrapper.html()).toMatchSnapshot();
  
});

//显示/隐藏排序
test('hidden sort', async () => {
  const wrapper = shallowMount(tableHeader, {
    props: {
      columns : [{
        key: 'id',
        label: 'ID',
        sortable: false,
      }],
      sortParams :  {
        field: '',
        direction: '',
      },
    },
  })
  expect(wrapper.find('.table-header-sort').exists() === false).toBe(true)

  await wrapper.setProps({
    columns : [{
      key: 'id',
      label: 'ID',
      sortable: true,
    }],
  })

  expect(wrapper.find('.table-header-sort').exists() === true).toBe(true)
})

// 点击排序
test('test click sort', async () => {
  const wrapper = shallowMount(tableHeader, {
    props: {
      columns : COLUMN,
      sortParams :  {
        field: '',
        direction: '',
      },
    },
  })
  const sortColumn = wrapper.find('.table-header-sort')
  //升序
  sortColumn.trigger('click')
  await nextTick()
  expect(wrapper.props().sortParams.field === 'id' && wrapper.props().sortParams.direction === 'ASC').toBe(true)
  //升序
  sortColumn.trigger('click')
  await nextTick()
  expect(wrapper.props().sortParams.field === 'id' && wrapper.props().sortParams.direction === 'DESC').toBe(true)
  //恢复无序
  sortColumn.trigger('click')
  await nextTick()
  expect(wrapper.props().sortParams.field === 'id' && wrapper.props().sortParams.direction === '').toBe(true)
})

