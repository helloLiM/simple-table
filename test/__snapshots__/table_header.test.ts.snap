// Vitest Snapshot v1

exports[`mount component 1`] = `
"<thead class=\\"table-header\\">
  <th>ID<span class=\\"table-header-sort\\"><i class=\\"iconfont icon-caret-up table-header-sort-up\\"></i><i class=\\"iconfont icon-caret-down table-header-sort-down\\"></i></span></th>
  <th>名称<span class=\\"table-header-sort\\"><i class=\\"iconfont icon-caret-up table-header-sort-up\\"></i><i class=\\"iconfont icon-caret-down table-header-sort-down\\"></i></span></th>
  <th>性别</th>
  <th>年龄<span class=\\"table-header-sort\\"><i class=\\"iconfont icon-caret-up table-header-sort-up\\"></i><i class=\\"iconfont icon-caret-down table-header-sort-down\\"></i></span></th>
</thead>"
`;
